File: prototype/player.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Player:
    x: int
    y: int
    z: int
    yaw: float

    @property
    def position(self) -> tuple[int, int, int]:
        return (self.x, self.y, self.z)

    @property
    def scan_origin(self) -> tuple[int, int, int]:
        """
        Prototype scan origin.

        For now, we use the player's block position directly.
        This can later be adjusted if we want the scan to start
        from torso or head level more explicitly.
        """
        return (self.x, self.y, self.z)

File: java-core/src/main/java/com/minersecho/Player.java
package com.minersecho;

public final class Player {
    public final String playerId;
    public final int x;
    public final int y;
    public final int z;
    public final double yaw;

    public Player(String playerId, int x, int y, int z, double yaw) {
        this.playerId = playerId;
        this.x = x;
        this.y = y;
        this.z = z;
        this.yaw = yaw;
    }

    public Vec3 position() {
        return new Vec3(x, y, z);
    }

    public Vec3 scanOrigin() {
        return position();
    }
}
